oak_functions_launcher/src/lookup.rs

```diff
@@ -21,55 +21,70 @@ use crate::{
 use anyhow::{anyhow, Context};
 use hashbrown::HashMap;
 use prost::Message;
-use std::{fs, path::PathBuf, vec::IntoIter};
+use std::{fs, path::PathBuf};
 use ubyte::ByteUnit;
 
-struct UpdateClient<'a> {
+struct UpdateClient<'a, I: Iterator<Item = LookupDataChunk>> {
     inner: &'a mut OakFunctionsAsyncClient<ConnectorHandle>,
-    chunks: IntoIter<LookupDataChunk>,
+    chunks: I,
 }
 
-impl UpdateClient<'_> {
+impl<I: Iterator<Item = LookupDataChunk>> UpdateClient<'_, I> {
```

Comment:
This works, but it seems unnecessary to try and monomorphize the dispatch on the concrete type of the iterator via this generic. I think dynamic dispatch via `Box<dyn Iterator<...>>` would be perfectly acceptable and more readable, but up to you.

Reply:
Curiously using `chunks: Box<dyn Iterator<Item = LookupDataChunk>>` in `UpdateClient` leads to:


```
error: future cannot be sent between threads safely
[10:41:56 ✓:0,✗:0,⠇:1] ││ ║    --> oak_functions_launcher/src/lib.rs:79:22
[10:41:56 ✓:0,✗:0,⠇:1] ││ ║     |
[10:41:56 ✓:0,✗:0,⠇:1] ││ ║ 79  |         tokio::spawn(setup_periodic_update(client, config));
[10:41:56 ✓:0,✗:0,⠇:1] ││ ║     |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ future returned by `setup_periodic_update` is not `Send`
[10:41:56 ✓:0,✗:0,⠇:1] ││ ║     |
[10:41:56 ✓:0,✗:0,⠇:1] ││ ║     = help: the trait `std::marker::Send` is not implemented for `(dyn std::iter::Iterator<Item = schema::LookupDataChunk> + 'static)`
[10:41:56 ✓:0,✗:0,⠇:1] ││ ║ note: future is not `Send` as this value is used across an await
[10:41:56 ✓:0,✗:0,⠇:1] ││ ║    --> oak_functions_launcher/src/lookup.rs:114:41
[10:41:56 ✓:0,✗:0,⠇:1] ││ ║     |
[10:41:56 ✓:0,✗:0,⠇:1] ││ ║ 108 |     let mut update_client = UpdateClient {
[10:41:56 ✓:0,✗:0,⠇:1] ││ ║     |         ----------------- has type `lookup::UpdateClient<'_>` which is not `Send`
[10:41:56 ✓:0,✗:0,⠇:1] ││ ║ ...
[10:41:56 ✓:0,✗:0,⠇:1] ││ ║ 114 |         update_client.start_and_finish().await
[10:41:56 ✓:0,✗:0,⠇:1] ││ ║     |                                         ^^^^^^ await occurs here, with `mut update_client` maybe used later
[10:41:56 ✓:0,✗:0,⠇:1] ││ ║ ...
[10:41:56 ✓:0,✗:0,⠇:1] ││ ║ 118 | }
[10:41:56 ✓:0,✗:0,⠇:1] ││ ║     | - `mut update_client` is later dropped here
[10:41:56 ✓:0,✗:0,⠇:1] ││ ║ note: required by a bound in `tokio::spawn`
[10:41:56 ✓:0,✗:0,⠇:1] ││ ║    --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.25.0/src/task/spawn.rs:163:21
[10:41:56 ✓:0,✗:0,⠇:1] ││ ║     |
[10:41:56 ✓:0,✗:0,⠇:1] ││ ║ 163 |         T: Future + Send + 'static,
[10:41:56 ✓:0,✗:0,⠇:1] ││ ║     |                     ^^^^ required by this bound in `spawn`
[10:41:56 ✓:0,✗:0,⠇:1] ││ ║ 
[10:41:56 ✓:0,✗:0,⠇:1] ││ ║ error: could not compile `oak_functions_launcher` due to previous error
[10:41:56 ✓:0,✗:0,⠇:1] ││ ║ warning: build failed, waiting for other jobs to finish...
```

I am curious to dig into this more, but I'd do that in a future PR.